blockmanager.go

```diff
@@ -480,7 +488,6 @@ func (b *blockManager) handleTxMsg(tmsg *txMsg) {
 	// instances of trying to fetch it, or we failed to insert and thus
 	// we'll retry next time we get an inv.
 	txHash := tmsg.tx.Sha()
-	delete(tmsg.peer.requestedTxns, *txHash)
```

Comment:
This still need to be removed.  If the above advice is followed it would be:

```Go
if requestedTxns, ok := b.requestedTxns[p.ID()]; ok {
	delete(requestedTxns, *txHash)
}
```